File: console/src/types/edge_comparison.ts
export interface SourceOddsEntry {
  source: string
  prob: number
  best_bid: number | null
  best_ask: number | null
  bookmaker: string | null
}

export interface EdgeComparisonEntry {
  contest_id: string
  match: string
  market_type: string
  outcome: string
  model_prob: number | null
  polymarket_prob: number | null
  polymarket_volume: number | null
  oddspapi_prob: number | null
  oddspapi_bookmaker: string | null
  edge_vs_polymarket: number | null
  edge_vs_oddspapi: number | null
  edge_diff: number | null
  sources: SourceOddsEntry[]
}

export interface EdgeComparisonResponse {
  date: string
  comparisons: EdgeComparisonEntry[]
  markets_with_both_sources: number
  total_markets: number
}

File: api/models/edge_comparison.py
from __future__ import annotations

from pydantic import BaseModel


class SourceOddsEntry(BaseModel):
    source: str
    prob: float
    best_bid: float | None = None
    best_ask: float | None = None
    bookmaker: str | None = None


class EdgeComparisonEntry(BaseModel):
    contest_id: str
    match: str
    market_type: str
    outcome: str
    model_prob: float | None = None
    polymarket_prob: float | None = None
    polymarket_volume: float | None = None
    oddspapi_prob: float | None = None
    oddspapi_bookmaker: str | None = None
    edge_vs_polymarket: float | None = None
    edge_vs_oddspapi: float | None = None
    edge_diff: float | None = None
    sources: list[SourceOddsEntry] = []


class EdgeComparisonResponse(BaseModel):
    date: str
    comparisons: list[EdgeComparisonEntry]
    markets_with_both_sources: int
    total_markets: int
